utils/modular_model_converter.py

```diff
@@ -808,322 +1005,261 @@ def visit_ImportFrom(self, node: cst.ImportFrom) -> None:
+                                    elements=[cst.Element(value=cst.SimpleString(value=k)) for k in new_alls]
+                                )
+                            )
+                            self.all_all_to_add[file] = node.with_changes(body=[new_node])
 
-    def leave_FunctionDef(self, original_node, node):
-        parent_node = self.get_metadata(cst.metadata.ParentNodeProvider, original_node)
-        if m.matches(parent_node, m.Module()):
-            self.all_definitions[node.name.value] = node
-        return node
-
-    def visit_Assign(self, node: cst.Assign) -> None:
-        # Check if the assignment target is '__all__'
-        if isinstance(node.targets[0].target, cst.Name) and node.targets[0].target.value == "__all__":
-            if isinstance(node.value, cst.List):
-                # Extract the elements from the list
-                all_all_to_add = defaultdict(list)
-                for elt in node.value.elements:
-                    if isinstance(elt.value, cst.SimpleString):
-                        # Remove quotes and add the string to the elements list
-                        class_name = elt.value.value
-                        file = self.class_to_file_type[
-                            elt.value.evaluated_value
-                        ]  # evaluated value give the content of the string
-                        all_all_to_add[file] += [class_name]
-                for f_type, new_alls in all_all_to_add.items():
-                    updated_node = node.with_changes(
-                        value=cst.List(elements=[cst.Element(value=cst.SimpleString(value=k)) for k in new_alls])
-                    )
-                    self.files[f_type][class_name] = {
-                        "insert_idx": self.global_scope_index + 100,
-                        "node": updated_node,
-                    }
-
-    def leave_If(self, original_node, node):
-        parent_node = self.get_metadata(cst.metadata.ParentNodeProvider, original_node)
-        if m.matches(parent_node, m.Module()):
-            full_statement = self.python_module.code_for_node(original_node.test)
-            if re.search(r"[\s\S]*is_.*available", full_statement):
-                self.all_safe_imports.append(node)
-            elif full_statement not in self.all_imports:
-                logger.warning(f"one import is protected with `if`. Hard guess where it's used {full_statement}")
-        return node
+    def leave_Module(self, node):
+        """When we leave the modular file, we do the following in order:
+        - compute recursive function dependencies
+        - for each model-specific file found in the imports, rename it with the new model name, visit it, and update
+        its dependency graph with the new function and assignment definitions found in the modular
+        - update the modular dependency graph with the imported functions and assignments (found when visiting the matching files)
```

Comment:
```suggestion
        1. compute the nested function dependencies
        2. for each model-specific file found in the imports, rename it with the new model name, visit it, and update
        its dependency graph with the new function and assignment definitions found in the modular
        3. update the modular dependency graph with the imported functions and assignments (found when visiting the matching files)
```
would be nice if these are reflected with # 1. , # 2. etc as comments

Reply:
>  for each model-specific file found in the imports, rename it with the new model name: for this we parse the file, and use a FileModelMapper, to visit it and upsate the model name, dependency graph etc